lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -383,6 +383,23 @@ ::mlir::OpFoldResult mlir::tt::ttir::ReshapeOp::fold(FoldAdaptor adaptor) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// BroadcastOp
+//===----------------------------------------------------------------------===//
+
+// BroadcastOp verification
+::mlir::LogicalResult mlir::tt::ttir::BroadcastOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+
+  // Check that the input rank matches the rank of the output tensor
```

Comment:
I will have a subsequent PR that will add and use everything around TTIR_Broadcastable attribute, so want to keep those changes separate.

Reply:
Ok I read your comment again and now understand better what you suggested now. I tried to add `TTIR_Broadcastable` to the `BroadcastOp`, but it requires some custom changes in the `TTIROpsInterfaces.cpp` since `BroadcastOp` is treated as a `UnaryOp` that can validly broadcast, so I just copied the check for mlir `getBroadcastedShape()` to the verify function of `BroadcastOp`.